caps/capabilities.go

```diff
@@ -107,3 +118,35 @@ func (r *Repository) Names() []string {
 	sort.Strings(keys)
 	return keys
 }
+
+// String representation of a capability.
+func (c Capability) String() string {
+	return c.Name
+}
+
+type byName []Capability
+
+func (c byName) Len() int           { return len(c) }
+func (c byName) Swap(i, j int)      { c[i], c[j] = c[j], c[i] }
+func (c byName) Less(i, j int) bool { return c[i].Name < c[j].Name }
+
+// All returns all capabilities ordered by name.
+func (r *Repository) All() []Capability {
+	caps := make([]Capability, len(r.caps))
+	i := 0
+	for _, capability := range r.caps {
+		caps[i] = *capability
+		i++
+	}
+	sort.Sort(byName(caps))
+	return caps
+}
+
+func (e *NotFoundError) Error() string {
+	switch e.what {
```

Comment:
For future benefits, this might be something more trivial. E.g.:

    return fmt.Errorf("cannot %s capability %s: not found", e.what, e.name)